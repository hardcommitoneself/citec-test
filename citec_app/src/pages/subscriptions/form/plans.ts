// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
export const PlansInfo = [
   {
      plan: 'Basic',
      price: '19',
      button: 'Start free trial',
      list: [
         'Portfolio Analysis',
         'Portfolio Optimization',
         'Portfolio Rebalance',
         'Basic Stock Screener',
         'Reports',
         'Save upto 10 portfolios',
         'Integration with Broker',
         'One Click batch Trading',
         'Up to $100,000 in Assets',
      ],
      price_id: process.env.REACT_APP_id_BASIC_SUBSCRIPTION
   },
   {
      plan: 'Professional',
      price: '49',
      button: 'Coming Soon',
      list: [
         'Functionality in Basic',
         'Advanced Stock Screener',
         'ESG Analytics',
         'Portfolio Ideas',
         'Up to $1,000,000 in Assets',
         'Save upto 10 portfolios',
      ],
      price_id: 'Coming-soon'
   },
   {
      plan: 'Institutional',
      price: 'Custom',
      button: 'Contact us',
      list: [
         'Functionality in Professional',
         'Manage Multiple Accounts',
         'Customized Reporting',
         'Assigned Specialist',
         'Unlimited Assets ',
      ],
      price_id: 'Custom'
   },
]
