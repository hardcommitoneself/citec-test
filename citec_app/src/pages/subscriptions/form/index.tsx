// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
import { PlansInfo } from './plans'
import PlanBox from 'features/subscriptions/ui/plan-box'
import { PageContainer } from "features/ui/page-container";

export function StripeForm() {

   return (
      <PageContainer>
         <div className="PortfolioCreate Pricing">
            <section className="body_area  ">
               <div className="content common_width">
                  <h1>
                     Are you ready to start <span>Making Money</span>!
                  </h1>
                  <div className="boxes_wrapper">
                     {PlansInfo.map((Each) => (
                        <PlanBox
                           plan={Each.plan}
                           price={Each.price}
                           list={Each.list}
                           button={Each.button}
                           price_id={Each.price_id}
                        />
                     ))}
                  </div>
               </div>
            </section>
         </div>
      </PageContainer>

   );
}