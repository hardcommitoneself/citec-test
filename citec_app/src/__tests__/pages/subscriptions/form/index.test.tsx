// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import { StripeForm } from 'pages/subscriptions/form'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

const mockStore = configureStore()
const store = mockStore()

test('renders Stripe Page Plans', () => {
   const { container } = render(
      <Provider store={store}>
         <StripeForm />
      </Provider>
   )
   const linkElement = screen.getByText(/Functionality in Basic/i)
   expect(linkElement).toBeInTheDocument()
   expect(container).toMatchSnapshot()
})
