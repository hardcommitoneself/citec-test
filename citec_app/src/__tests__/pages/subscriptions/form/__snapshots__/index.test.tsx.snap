// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Stripe Page Plans 1`] = `
<div>
  <div
    class="undefined my-8 mx-10 undefined rounded-2xl relative"
    id="pagecontainer"
  >
    <div
      class="criteria_title flex justify-between"
    />
    <div
      class="PortfolioCreate Pricing"
    >
      <section
        class="body_area  "
      >
        <div
          class="content common_width"
        >
          <h1>
            Are you ready to start 
            <span>
              Making Money
            </span>
            !
          </h1>
          <div
            class="boxes_wrapper"
          >
            <div
              class="PricingBox false"
            >
              <h1
                class="heading"
              >
                Basic
              </h1>
              <h2
                class="price"
              >
                $
                19
                <span>
                  .99
                </span>
              </h2>
              <button>
                Start free trial
              </button>
              <ul>
                <li>
                  <img
                    alt=""
                    src="tick_list.png"
                  />
                  <p>
                    Portfolio Analysis
                  </p>
                </li>
                <li>
                  <img
                    alt=""
                    src="tick_list.png"
                  />
                  <p>
                    Portfolio Optimization
                  </p>
                </li>
                <li>
                  <img
                    alt=""
                    src="tick_list.png"
                  />
                  <p>
                    Portfolio Rebalance
                  </p>
                </li>
                <li>
                  <img
                    alt=""
                    src="tick_list.png"
                  />
                  <p>
                    Basic Stock Screener
                  </p>
                </li>
                <li>
                  <img
                    alt=""
                    src="tick_list.png"
                  />
                  <p>
                    Reports
                  </p>
                </li>
                <li>
                  <img
                    alt=""
                    src="tick_list.png"
                  />
                  <p>
                    Save upto 10 portfolios
                  </p>
                </li>
                <li>
                  <img
                    alt=""
                    src="tick_list.png"
                  />
                  <p>
                    Integration with Broker
                  </p>
                </li>
                <li>
                  <img
                    alt=""
                    src="tick_list.png"
                  />
                  <p>
                    One Click batch Trading
                  </p>
                </li>
                <li>
                  <img
                    alt=""
                    src="tick_list.png"
                  />
                  <p>
                    Up to $100,000 in Assets
                  </p>
                </li>
              </ul>
            </div>
            <div
              class="PricingBox false"
            >
              <h1
                class="heading"
              >
                Professional
              </h1>
              <h2
                class="price"
              >
                $
                49
                <span>
                  .99
                </span>
              </h2>
              <button>
                Coming Soon
              </button>
              <ul>
                <li>
                  <img
                    alt=""
                    src="tick_list.png"
                  />
                  <p>
                    Functionality in Basic
                  </p>
                </li>
                <li>
                  <img
                    alt=""
                    src="tick_list.png"
                  />
                  <p>
                    Advanced Stock Screener
                  </p>
                </li>
                <li>
                  <img
                    alt=""
                    src="tick_list.png"
                  />
                  <p>
                    ESG Analytics
                  </p>
                </li>
                <li>
                  <img
                    alt=""
                    src="tick_list.png"
                  />
                  <p>
                    Portfolio Ideas
                  </p>
                </li>
                <li>
                  <img
                    alt=""
                    src="tick_list.png"
                  />
                  <p>
                    Up to $1,000,000 in Assets
                  </p>
                </li>
                <li>
                  <img
                    alt=""
                    src="tick_list.png"
                  />
                  <p>
                    Save upto 10 portfolios
                  </p>
                </li>
              </ul>
            </div>
            <div
              class="PricingBox false"
            >
              <h1
                class="heading"
              >
                Institutional
              </h1>
              <h2
                class="price CustomPrice "
              >
                Custom
              </h2>
              <button>
                Contact us
              </button>
              <ul>
                <li>
                  <img
                    alt=""
                    src="tick_list.png"
                  />
                  <p>
                    Functionality in Professional
                  </p>
                </li>
                <li>
                  <img
                    alt=""
                    src="tick_list.png"
                  />
                  <p>
                    Manage Multiple Accounts
                  </p>
                </li>
                <li>
                  <img
                    alt=""
                    src="tick_list.png"
                  />
                  <p>
                    Customized Reporting
                  </p>
                </li>
                <li>
                  <img
                    alt=""
                    src="tick_list.png"
                  />
                  <p>
                    Assigned Specialist
                  </p>
                </li>
                <li>
                  <img
                    alt=""
                    src="tick_list.png"
                  />
                  <p>
                    Unlimited Assets 
                  </p>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;
